tcms/testcases/migrations/0011_trim_bugsystem_fields.py

```diff
@@ -10,6 +10,20 @@ class Migration(migrations.Migration):
     ]
 
     operations = [
+        migrations.AlterField(model_name='bugsystem',
+                              name='validate_reg_exp',
+                              field=models.CharField(max_length=128,
+                                                     help_text='A valid JavaScript regular '
+                                                               'expression such as ^\\d$',
+                                                     verbose_name='RegExp for ID validation',
+                                                     default='')),
+        migrations.AlterField(model_name='bugsystem',
+                              name='url_reg_exp',
+                              field=models.CharField(max_length=8192,
+                                                     help_text='A valid Python format string such '
+                                                               'as http://bugs.example.com/%s',
+                                                     verbose_name='URL format string',
+                                                     default='')),
```

Comment:
This is not the proper way to resolve this b/c in case of rollback it leaves the DB in n inconsistent state compared to the underlying models. I think the following will work, but haven't tried it out

1) Add `preserve_default=False` parameter to `AlterField`, see https://docs.djangoproject.com/en/3.0/ref/migration-operations/#alterfield - this should make the fields nullable only for the current transaction without actually altering the DB schema. This may need to actually be placed at the end of the operations list or maybe not even needed once you have the RunPython piece mentioned below.

2) Before removing any of the fields use `RunPython` and dump all values to disk in the forwards function
3) In the backwards function read the values from disk and update these fields - could be json.loads + store the entire object directly I think or you can play with `.save()`'s `update_fields` parameter, see https://docs.djangoproject.com/en/3.1/ref/models/instances/#django.db.models.Model.save

Reply:
I had already tried the dumping & restoring the data in the RunPython operation. Maybe try it out, because the `preserve_default=False` option isn't helping as we're back to the same problem as before.

Reply:
preserve_default will also need a default value in the field definition itself.